### DISCUSSION TITLE
Add class to fancybox__content element?

### DISCUSSION THREAD

**Discussion by LewBlu** [fancyapps/ui#349] [Q&A]
Hi, 

I've tried adding a class to the element that is created with "fancybox__content" class on it but unable to get it to work. Below is the code that I tried to use:
`
Fancybox.show([{
			src: '/example',
			type: "ajax",
			dragToClose: false,
			keyboard: false,
			mainClass: "exampleClass"
		}]);`

I need to be able to add this class so that I can manage the width of the fancybox content depending on the content that I would like to show. I thought that the option to do this was "mainClass" but this doesn't seem to do anything?

Any help much appreciated.

**Comment by mDesignHub** [comment]
i would love to know this too

**Comment by fancyapps** [comment]
Hi, You can, for example, simply use callbacks to access slide element, and
then you can do whatever you need.

Janis

**Reply by mDesignHub** [reply]
thx janis. could you give us an example code syntax for better understanding.

**Reply by fancyapps** [reply]
Example:

```JS
Fancybox.show(
  [
    {
      src: "/example",
      type: "ajax",
      dragToClose: false,
      keyboard: false,
    },
  ],
  {
    on: {
      load: (_fancybox, slide) => {
        slide.$el.classList.add("exampleClass");
      },
    },
  }
);

```

**Reply by mDesignHub** [reply]
Thx janis, this helped alot,
could you tell me the correct syntax for adding a class only to div.fancybox__content ?

**Reply by fancyapps** [reply]
Simply check out `slide` variable, it contains useful info, for example, `$content` is a reference to the wrapping element.

**Reply by pork-chop** [reply]
I am trying to use the same demo to create a custom pop up that fills with dynamic data. I've tried quite a bit, but I cannot target `slide` as a variable - it just keeps returning _undefined_.

I can populate `$leftCol` and `$rightCol` exactly as I want, and I've got my CSS all set up so they're styled correctly. But I can't figure out how to move these elements in the DOM into the stage of `div.fancybox__content`. 

_Update -- I figured this out with a help of a colleague. I was trying to access 'slide' in a lifecycle stage that didn't quite exist in yet. I needed to work in the "done" stage._
